pkg/issuer/acme/dns/clouddns/clouddns.go

```diff
@@ -217,7 +217,13 @@ func (c *DNSProvider) getHostedZone(domain string) (string, error) {
 		return "", fmt.Errorf("No matching GoogleCloud domain found for domain %s", authZone)
 	}
 
-	return zones.ManagedZones[0].Name, nil
+	for _, zone := range zones.ManagedZones {
+		if zone.Visibility == "public" {
+			return zone.Name, nil
+		}
```

Comment:
If none of the zones are public but a private zone is found, can we fall back to using this anyway and maybe log a debug/info message?

Reply:
i.e. fall-back to the 0th if a public one isn't found?